### DISCUSSION TITLE
Exception e0000008, but app is not BrowserSubprocess

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3618] [General]
While testing my app I get app crash with Cefsharp Offscreen version. In debug mode it also showing me this window:

![image](https://user-images.githubusercontent.com/73235464/122022448-438a6600-cdcf-11eb-8ffa-bcd9f37cecf2.png)

I suppose, it can be related to out of memory exception. I just noticed, that exception code is "e0000008", however app name is MYAPP and module with error is kernelbase.dll.

![image](https://user-images.githubusercontent.com/73235464/122022710-877d6b00-cdcf-11eb-8639-9d06c18a0cb0.png)

I found that e0000008 - is typical exception for BrowserSubprocess, so could that be somehow related with Cefsharp? Any ideas how to go deeper?

My OS is Windows7 64x , VS 2019 installed. I use Offscreen Cefsharp version, .NET 4.5.2 . Cef version is 90.6.50.0

**Comment by amaitland** [comment]
Is your application x64 or x86? This is often seen when the process runs out of memory. Most common when running  32bit applications (x86). 32bit applications need to be made large address aware. https://github.com/cefsharp/CefSharp/wiki/General-Usage#win32-out-of-memory

You can download the libcef.dll.pdb if the call stack originates from libcef
https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#loading-native-symbols-for-easier-diagnosis

I'd suggest enabling crash pad if you haven't already see https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#crash-reporting
Once you have a mini dump you can get a stack trace.

**Reply by assopri** [reply]
Switching to x64 platform helped to avoid this error. However I`m still getting another one:
![image](https://user-images.githubusercontent.com/73235464/122797919-ab0d4d80-d2c8-11eb-8090-850d53f8443c.png)

Which pdb file should I download to get its call stack? (as far as I seem it is not originated by libcef)

**Reply by assopri** [reply]
One more thing - where can I find pdb files for 90.6.50.0 version?

**Reply by amaitland** [reply]
Download the libcef.dll.pdb from https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#loading-native-symbols-for-easier-diagnosis

If the Browsersubprocess is crashing it's most likely that libcef.dll is the source of the crash. If you've enabled crash reporting then I don't believe you should be seeing the dialog box, double check you've enabled it correctly by checking the log file.

The CefSharp pdbs are included in the Nuget package, they should be copied to your bin folder by default, you can manually copy them if required.

**Reply by assopri** [reply]
Ok, to enable crash logs I added crash_reporter.cfg like this to the exe folder:
![image](https://user-images.githubusercontent.com/73235464/122882604-0a10a800-d345-11eb-9cbb-b8932a843df5.png)

Got dump in  C:\Users\[CurrentUser]\AppData\Local\[AppName]\User Data folder and no dialog box:
![image](https://user-images.githubusercontent.com/73235464/122883255-adfa5380-d345-11eb-8506-70eede561ff6.png)

Still haven't found libcef.dll.pdb on my computer (searched with Everything app). How can I get this file (maybe somehow download from here https://www.nuget.org/packages/cef.redist.x64/90.6.5 ) ? I downloaded cef.redist.x64.90.6.5.nupkg , still not aware how can I extract pdb from package.

**Reply by amaitland** [reply]
> Still haven't found libcef.dll.pdb on my computer (searched with Everything app). 

The instructions for locating and downloading libcef.dll.pdb are at https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#loading-native-symbols-for-easier-diagnosis

Make sure to download the release symbols.

**Reply by assopri** [reply]
Ok thank you! I found the corresponding libcef.dll.pdb
![image](https://user-images.githubusercontent.com/73235464/123091523-20486200-d432-11eb-8fdd-19478392b63a.png)

then put it to exe folder and again launched app to arose bug. And again I got the same dump:
![image](https://user-images.githubusercontent.com/73235464/123091706-5685e180-d432-11eb-8f94-440d60d8e762.png)

How can I analyze it now to get more details on what exactly causes the crash?

p.s. by the way - the crash is causing on my app version just after processing it with ilprotector+smartassembly. Any ideas how could that happen?

**Reply by amaitland** [reply]
> How can I analyze it now to get more details on what exactly causes the crash?

You can load the mini dump in visual studio. 

> the crash is causing on my app version just after processing it with ilprotector+smartassembly. Any ideas how could that happen?

I don't know what you mean by this. I've never heard of ilprotector+smartassembly

**Reply by amaitland** [reply]
Basic instructions at https://docs.microsoft.com/en-us/visualstudio/debugger/using-dump-files?view=vs-2019#BKMK_Open_a_dump_file

Make sure to load the pdb file.

**Reply by assopri** [reply]
Ok, I had 4 files from the particular crash:
![image](https://user-images.githubusercontent.com/73235464/123238462-2d735880-d4e7-11eb-88c2-93200a2e4c6e.png)

I loaded the first one and got the following screen:
![image](https://user-images.githubusercontent.com/73235464/123238628-4bd95400-d4e7-11eb-8278-e39032036abf.png)

The other dumps are looking very similar.

What can I do to find out more details? How can I make sure that I loaded pdb file, except for just checking its presence in exe folder of my app?

![image](https://user-images.githubusercontent.com/73235464/123238920-8fcc5900-d4e7-11eb-8a38-bdcedc202d66.png)

**Reply by amaitland** [reply]
Try instructions in https://michaelscodingspot.com/how-to-create-use-and-debug-net-application-crash-dumps-in-2019/#Debug-With-Visual-Studio

**Reply by assopri** [reply]
Thank you for your step by step help! I appreciate it!
